### DISCUSSION TITLE
Crash when modifing vrp-reload.py

### DISCUSSION THREAD

**Discussion by orangiee** [google/or-tools#2906] [Routing (and legacy CP) questions] [answered]
**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
vrp-reload.py

**What operating system (Linux, Windows, ...) and version?**
macOS Big Sur, version 11.6.1

When i ran the script cvrp_reload.py which was modified and added more data(163 nodes with time windows) at Terminal, i got an information: Segmentation fault: 11, and all information like this: 

Process:               Python [84579]
Path:                  /usr/local/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python
Identifier:            Python
Version:               3.9.8 (3.9.8)
Code Type:             X86-64 (Native)
Parent Process:        bash [2913]
Responsible:           Terminal [2743]
User ID:               501

Date/Time:             2021-11-19 16:47:43.771 +0800
OS Version:            macOS 11.6.1 (20G224)
Report Version:        12
Bridge OS Version:     6.0 (19P549)
Anonymous UUID:        420B06B0-3E86-EC63-4D9E-308FD106956E

Sleep/Wake UUID:       6B63DB16-B59F-48EB-886C-887887581BD7

Time Awake Since Boot: 4000 seconds
Time Since Wake:       620 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       EXC_I386_GPFLT
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [84579]

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   _pywrapcp.so                  	0x0000000103247ba2 _wrap_RoutingDimension_SlackVar(_object*, _object*) + 258
1   org.python.python             	0x0000000102b13ad3 cfunction_call + 90
2   org.python.python             	0x0000000102ac813a _PyObject_MakeTpCall + 129
3   org.python.python             	0x0000000102bc1437 call_function + 279
4   org.python.python             	0x0000000102bbaae3 _PyEval_EvalFrameDefault + 27880
5   org.python.python             	0x0000000102ac8b93 _PyFunction_Vectorcall + 192
6   org.python.python             	0x0000000102bc13c8 call_function + 168
7   org.python.python             	0x0000000102bb9b20 _PyEval_EvalFrameDefault + 23845
8   org.python.python             	0x0000000102ac8b93 _PyFunction_Vectorcall + 192
9   org.python.python             	0x0000000102bc13c8 call_function + 168
10  org.python.python             	0x0000000102bb8e05 _PyEval_EvalFrameDefault + 20490
11  org.python.python             	0x0000000102ac8b93 _PyFunction_Vectorcall + 192
12  org.python.python             	0x0000000102bc13c8 call_function + 168
13  org.python.python             	0x0000000102bb8e05 _PyEval_EvalFrameDefault + 20490
14  org.python.python             	0x0000000102bb2d01 _PyEval_EvalCode + 411
15  org.python.python             	0x0000000102c0e1bd run_eval_code_obj + 128
16  org.python.python             	0x0000000102c0e10b run_mod + 96
17  org.python.python             	0x0000000102c0b939 pyrun_file + 167
18  org.python.python             	0x0000000102c0b320 pyrun_simple_file + 271
19  org.python.python             	0x0000000102c0b1eb PyRun_SimpleFileExFlags + 67
20  org.python.python             	0x0000000102c2ae8a pymain_run_file + 326
21  org.python.python             	0x0000000102c2a6ce Py_RunMain + 990
22  org.python.python             	0x0000000102c2b790 pymain_main + 35
23  org.python.python             	0x0000000102c2ba66 Py_BytesMain + 42
24  libdyld.dylib                 	0x00007fff2045df3d start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x00007fef6c054870  rbx: 0x00007fef6d044c00  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0xfffffffffffffffb  rsi: 0xfffffffffffffffe  rbp: 0x00007ffeed1a4a10  rsp: 0x00007ffeed1a49e0
   r8: 0x0000000000000002   r9: 0x000000000000007f  r10: 0x00007ffeed1a49b8  r11: 0x0000000000000001
  r12: 0x0000000000000000  r13: 0x0000000103097980  r14: 0x00007fef6c04c790  r15: 0x0000000104ece360
  rip: 0x0000000103247ba2  rfl: 0x0000000000010246  cr2: 0x00007fef6c072050
  
Logical CPU:     2
Error Code:      0x00000000
Trap Number:     13

Thread 0 instruction stream:
  00 48 8b 04 c1 eb 35 4c-8b 75 d8 e8 c4 2c 01 00  .H....5L.u...,..
  48 89 c3 e8 62 2c 01 00-48 85 c0 74 3c e8 40 2c  H...b,..H..t<.@,
  01 00 48 8b 05 bd b4 07-00 48 8b 38 48 8d 35 37  ..H......H.8H.57
  30 07 00 eb 11 48 8b 05-b2 b4 07 00 48 8b 38 48  0....H......H.8H
  8d 35 ba 2f 07 00 e8 3b-2c 01 00 31 db 48 89 d8  .5./...;,..1.H..
  48 83 c4 20 5b 41 5e 5d-c3 49 8b 86 00 02 00 00  H.. [A^].I......
 [48]8b 3c d8 48 8b 35 83-d0 08 00 31 d2 e8 8c b8  H.<.H.5....1....	<==
  f5 ff 48 89 c3 eb d6 0f-1f 80 00 00 00 00 55 48  ..H...........UH
  89 e5 41 57 41 56 53 48-83 ec 28 48 89 f7 48 c7  ..AWAVSH..(H..H.
  45 c8 00 00 00 00 48 c7-45 d0 00 00 00 00 48 c7  E.....H.E.....H.
  45 d8 00 00 00 00 48 c7-45 e0 00 00 00 00 48 8d  E.....H.E.....H.
  45 e0 48 89 04 24 48 8d-35 c0 f4 03 00 31 db 4c  E.H..$H.5....1.L
  
Thread 0 last branch register state not available.

Binary Images:
       0x102a5a000 -        0x102a5dfff +org.python.python (3.9.8 - 3.9.8) <D86D3FBF-68F2-342B-9928-B22F0E108C42> /usr/local/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python
       0x102a67000 -        0x102d2afff +org.python.python (3.9.8, [c] 2001-2021 Python Software Foundation. - 3.9.8) <971AA2A9-9678-3B45-9D18-D97B6CA2A0B6> /usr/local/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/Python
       0x10300f000 -        0x103012fff +_heapq.cpython-39-darwin.so (0) <CC37EE1C-A51F-32DA-A26F-CC229E866FF1> /usr/local/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload/_heapq.cpython-39-darwin.so
       0x10319f000 -        0x1032c2fff +_pywrapcp.so (0) <F4AB7087-DBEF-3EB5-B974-690CCD75F2F9> /Users/USER/Library/Python/3.9/lib/python/site-packages/ortools/constraint_solver/_pywrapcp.so
       0x103357000 -        0x10481afff +libortools.dylib (0) <EAF31027-0DE9-3EC8-97D3-48F9D3F96D62> /Users/USER/Library/Python/3.9/lib/python/site-packages/ortools/.libs/libortools.dylib
       0x104f7f000 -        0x104f7ffff +_api_implementation.cpython-39-darwin.so (0) <38FCEF85-8D81-3998-B478-0841AA48B92C> /Users/USER/Library/Python/3.9/lib/python/site-packages/google/protobuf/internal/_api_implementation.cpython-39-darwin.so
       0x104f82000 -        0x104f89fff +binascii.cpython-39-darwin.so (0) <31F6A6BE-D2DB-3471-933C-DB57B0F9EFF2> /usr/local/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload/binascii.cpython-39-darwin.so
       0x104f96000 -        0x1050f2fff +_message.cpython-39-darwin.so (0) <621938AE-4F54-3147-99DC-B21EB1997052> /Users/USER/Library/Python/3.9/lib/python/site-packages/google/protobuf/pyext/_message.cpython-39-darwin.so
       0x1052c9000 -        0x1052d0fff +_struct.cpython-39-darwin.so (0) <7404C151-1BAB-3F0E-A689-AB5F6C740A8A> /usr/local/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload/_struct.cpython-39-darwin.so
       0x1052dd000 -        0x1052f4fff +_pickle.cpython-39-darwin.so (0) <4581C00A-21EB-3BEA-A9C4-8D3987BA7B95> /usr/local/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload/_pickle.cpython-39-darwin.so
       0x105385000 -        0x105390fff +math.cpython-39-darwin.so (0) <ACE49B78-4BE1-3085-BCD8-D7CD6DF34B40> /usr/local/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload/math.cpython-39-darwin.so
       0x10539d000 -        0x1053acfff +_datetime.cpython-39-darwin.so (0) <7F2666AA-3CFF-3DCA-9D4F-6D016AB48C63> /usr/local/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload/_datetime.cpython-39-darwin.so
       0x10e958000 -        0x10e9f3fff  dyld (852.2) <14C5D17F-5032-37A2-9CD8-F437EB4A3992> /usr/lib/dyld
    0x7fff2017a000 -     0x7fff2017bfff  libsystem_blocks.dylib (79) <4DF6D8F5-D9C2-3A96-8DE4-5E99D6B73DC8> /usr/lib/system/libsystem_blocks.dylib
    0x7fff2017c000 -     0x7fff201b1fff  libxpc.dylib (2038.120.1) <0397FC9F-BD11-31FC-882E-9EDAA1E5CA65> /usr/lib/system/libxpc.dylib
    0x7fff201b2000 -     0x7fff201c9fff  libsystem_trace.dylib (1277.120.1) <0A24EB90-5396-36B0-A7E6-E9288702856D> /usr/lib/system/libsystem_trace.dylib
    0x7fff201ca000 -     0x7fff20267fff  libcorecrypto.dylib (1000.140.4) <57E7471E-3960-3398-8918-20DF446EA99B> /usr/lib/system/libcorecrypto.dylib
    0x7fff20268000 -     0x7fff20294fff  libsystem_malloc.dylib (317.140.5) <10C22FD0-FC7B-3325-852E-FEC4E88D2BC5> /usr/lib/system/libsystem_malloc.dylib
    0x7fff20295000 -     0x7fff202d9fff  libdispatch.dylib (1271.120.2) <BA7AD614-F2C2-3E89-9043-43DD548AE5B1> /usr/lib/system/libdispatch.dylib
    0x7fff202da000 -     0x7fff20313fff  libobjc.A.dylib (824) <6C842797-3EA9-30B4-A36D-E948DAF0EC89> /usr/lib/libobjc.A.dylib
    0x7fff20314000 -     0x7fff20316fff  libsystem_featureflags.dylib (28.60.1) <6EB33926-8E22-3000-B2F1-C6182B8CBD8F> /usr/lib/system/libsystem_featureflags.dylib
    0x7fff20317000 -     0x7fff2039ffff  libsystem_c.dylib (1439.141.1) <8447A4B8-0751-3EF1-AA9B-042E40EFA07D> /usr/lib/system/libsystem_c.dylib
    0x7fff203a0000 -     0x7fff203f5fff  libc++.1.dylib (905.6) <81674ADB-670F-3B19-AF5D-F3F66CB93D44> /usr/lib/libc++.1.dylib
    0x7fff203f6000 -     0x7fff2040bfff  libc++abi.dylib (905.6) <78CE7863-E224-3D0B-98F1-E5E3E382322D> /usr/lib/libc++abi.dylib
    0x7fff2040c000 -     0x7fff2043bfff  libsystem_kernel.dylib (7195.141.8) <8DFE5B9F-B718-3D48-ABB1-238122989218> /usr/lib/system/libsystem_kernel.dylib
    0x7fff2043c000 -     0x7fff20447fff  libsystem_pthread.dylib (454.120.2) <49670AEC-4D5D-3383-906C-23F568351FCB> /usr/lib/system/libsystem_pthread.dylib
    0x7fff20448000 -     0x7fff20483fff  libdyld.dylib (852.2) <5FBD0E1A-ACCE-36DB-B11C-622F26C85132> /usr/lib/system/libdyld.dylib
    0x7fff20484000 -     0x7fff2048dfff  libsystem_platform.dylib (254.80.2) <A85D12B6-6213-34EE-84D8-8E993C19E330> /usr/lib/system/libsystem_platform.dylib
    0x7fff2048e000 -     0x7fff204b9fff  libsystem_info.dylib (542.40.3) <36329FC6-9982-306A-87F9-D018E7F49F4B> /usr/lib/system/libsystem_info.dylib
    0x7fff204ba000 -     0x7fff20957fff  com.apple.CoreFoundation (6.9 - 1778.101) <CE026405-2DAF-39D2-A601-743B3A61366E> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fff22629000 -     0x7fff22888fff  libicucore.A.dylib (66112) <251FC896-7EA1-3505-A9AA-80BAE7A6FD2B> /usr/lib/libicucore.A.dylib
    0x7fff22889000 -     0x7fff22892fff  libsystem_darwin.dylib (1439.141.1) <29F82ABE-E1A0-3BC2-B91E-ADC010CF23FA> /usr/lib/system/libsystem_darwin.dylib
    0x7fff22ca8000 -     0x7fff22cb3fff  libsystem_notify.dylib (279.40.4) <608B5A07-D31A-3BEC-86BF-C2E498C085AF> /usr/lib/system/libsystem_notify.dylib
    0x7fff24c40000 -     0x7fff24c4efff  libsystem_networkextension.dylib (1295.140.3) <022BE26C-C058-3CC7-8E0B-348B3D3B639C> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff24cad000 -     0x7fff24cc3fff  libsystem_asl.dylib (385) <47B99A49-749E-356F-9D81-6D3A0B1544AB> /usr/lib/system/libsystem_asl.dylib
    0x7fff2638a000 -     0x7fff26391fff  libsystem_symptoms.dylib (1431.140.1) <EA6435E7-8F85-315B-8AED-C20A07DE7F96> /usr/lib/system/libsystem_symptoms.dylib
    0x7fff28399000 -     0x7fff283a9fff  libsystem_containermanager.dylib (318.100.4) <E0A89571-6D3E-3184-9F39-C6094C87B92B> /usr/lib/system/libsystem_containermanager.dylib
    0x7fff290ad000 -     0x7fff290b0fff  libsystem_configuration.dylib (1109.140.1) <89E70992-616F-3DD3-9430-67025F759A1B> /usr/lib/system/libsystem_configuration.dylib
    0x7fff290b1000 -     0x7fff290b5fff  libsystem_sandbox.dylib (1441.141.4) <087B0ADD-EF39-3E80-800F-F111FD65ABBA> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff29da7000 -     0x7fff29da9fff  libquarantine.dylib (119.40.2) <B1418CB0-CDC7-3EB2-ABC3-8609ADC92BC4> /usr/lib/system/libquarantine.dylib
    0x7fff2a352000 -     0x7fff2a356fff  libsystem_coreservices.dylib (127.1) <7AE405FC-6A44-34A0-86AD-6BD80B0050BB> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff2a553000 -     0x7fff2a565fff  libz.1.dylib (76) <73C471D3-07CE-307D-A531-92C5D0E4C0E9> /usr/lib/libz.1.dylib
    0x7fff2a566000 -     0x7fff2a5adfff  libsystem_m.dylib (3186.100.3) <EA0354A3-8618-3D76-A760-E550AC60CE95> /usr/lib/system/libsystem_m.dylib
    0x7fff2a5af000 -     0x7fff2a5b4fff  libmacho.dylib (980) <1FCE2BE3-4F6F-3EAA-9BC5-A9892A45CF0D> /usr/lib/system/libmacho.dylib
    0x7fff2a5d1000 -     0x7fff2a5dcfff  libcommonCrypto.dylib (60178.120.3) <7E242F29-1CB6-30EF-8C9A-C768A90FDBA0> /usr/lib/system/libcommonCrypto.dylib
    0x7fff2a5dd000 -     0x7fff2a5e7fff  libunwind.dylib (201) <A5B040A8-847F-36EE-B13D-5DD1F5CD5BED> /usr/lib/system/libunwind.dylib
    0x7fff2a5e8000 -     0x7fff2a5effff  liboah.dylib (203.58) <26D08622-69F5-32DB-80D2-9B4651A9F0CC> /usr/lib/liboah.dylib
    0x7fff2a5f0000 -     0x7fff2a5fafff  libcopyfile.dylib (173.40.2) <8C783785-0F5F-3DC5-B815-B29CEBA75737> /usr/lib/system/libcopyfile.dylib
    0x7fff2a5fb000 -     0x7fff2a602fff  libcompiler_rt.dylib (102.2) <1FDC92D1-8A17-30AF-8E72-4F0517AEA157> /usr/lib/system/libcompiler_rt.dylib
    0x7fff2a603000 -     0x7fff2a605fff  libsystem_collections.dylib (1439.141.1) <11D5775A-AD4C-35ED-BC05-616AB67ACBBE> /usr/lib/system/libsystem_collections.dylib
    0x7fff2a606000 -     0x7fff2a608fff  libsystem_secinit.dylib (87.60.1) <27982311-637E-3308-9F55-4871762736F4> /usr/lib/system/libsystem_secinit.dylib
    0x7fff2a609000 -     0x7fff2a60bfff  libremovefile.dylib (49.120.1) <5973CED7-797B-3288-9589-C1856752F91A> /usr/lib/system/libremovefile.dylib
    0x7fff2a60c000 -     0x7fff2a60cfff  libkeymgr.dylib (31) <AAA929A0-45E5-3186-8ABD-37EB25B2C939> /usr/lib/system/libkeymgr.dylib
    0x7fff2a60d000 -     0x7fff2a614fff  libsystem_dnssd.dylib (1310.140.1) <4A85F13B-E3AD-3C44-B327-091F56D35CC1> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff2a615000 -     0x7fff2a61afff  libcache.dylib (83) <435668CB-666B-3379-AD65-F604A72099F4> /usr/lib/system/libcache.dylib
    0x7fff2a61b000 -     0x7fff2a61cfff  libSystem.B.dylib (1292.120.1) <2553EE0F-3C42-3ACE-9400-EF055BD5BE8E> /usr/lib/libSystem.B.dylib
    0x7fff2a61d000 -     0x7fff2a620fff  libfakelink.dylib (3) <C7E36B51-4FD6-3704-975E-AB9B46DF6D30> /usr/lib/libfakelink.dylib
    0x7fff2a621000 -     0x7fff2a621fff  com.apple.SoftLinking (1.0 - 1) <AF16BDE9-8995-39F5-A0ED-C65232A61095> /System/Library/PrivateFrameworks/SoftLinking.framework/Versions/A/SoftLinking
    0x7fff2da1f000 -     0x7fff2da1ffff  liblaunch.dylib (2038.120.1) <A3F46CC6-BD89-39DB-8732-C885B881A635> /usr/lib/system/liblaunch.dylib
    0x7fff2feb6000 -     0x7fff2feb6fff  libsystem_product_info_filter.dylib (8.40.1) <6CA8DEA4-5BD4-375F-9AA7-3338135306C5> /usr/lib/system/libsystem_product_info_filter.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=547.2M resident=0K(0%) swapped_out_or_unallocated=547.2M(100%)
Writable regions: Total=80.5M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=80.5M(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Kernel Alloc Once                    8K        1 
MALLOC                            57.6M       36 
MALLOC guard page                   16K        4 
MALLOC_LARGE (reserved)            896K        7         reserved VM address space (unallocated)
STACK GUARD                          4K        1 
Stack                             16.0M        1 
VM_ALLOCATE                       5640K       24 
__DATA                            1478K       62 
__DATA_CONST                      3127K       48 
__DATA_DIRTY                        95K       24 
__LINKEDIT                       509.0M       24 
__OBJC_RO                         70.3M        1 
__OBJC_RW                         2496K        2 
__TEXT                            38.3M       60 
__UNICODE                          588K        1 
shared memory                        8K        2 
===========                     =======  ======= 
TOTAL                            705.1M      298 
TOTAL, minus reserved VM space   704.2M      298 

Model: MacBookPro15,1, BootROM 1715.40.15.0.0 (iBridge: 19.16.10549.0.0,0), 8 processors, 8-Core Intel Core i9, 2.3 GHz, 16 GB, SMC 
Graphics: kHW_IntelUHDGraphics630Item, Intel UHD Graphics 630, spdisplays_builtin
Graphics: kHW_AMDRadeonPro560XItem, Radeon Pro 560X, spdisplays_pcie_device, 4 GB
Memory Module: BANK 0/ChannelA-DIMM0, 8 GB, DDR4, 2400 MHz, Micron, 8ATF1G64HZ-2G6E1
Memory Module: BANK 2/ChannelB-DIMM0, 8 GB, DDR4, 2400 MHz, Micron, 8ATF1G64HZ-2G6E1
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x7BF), wl0: Aug 10 2021 19:55:28 version 9.30.444.18.32.5.71 FWID 01-93db3422
Bluetooth: Version 8.0.5d7, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB 3.1 Bus
USB Device: Apple T2 Bus
USB Device: Touch Bar Backlight
USB Device: Touch Bar Display
USB Device: Apple Internal Keyboard / Trackpad
USB Device: Headset
USB Device: Ambient Light Sensor
USB Device: FaceTime HD Camera (Built-in)
USB Device: Apple T2 Controller
Thunderbolt Bus: MacBook Pro, Apple Inc., 47.5
Thunderbolt Bus: MacBook Pro, Apple Inc., 47.5

Could anyone help me to read and interpret this ? Thanks in advance !

**Comment by lperron** [comment]
model is wrong. A few assumptions in the model are missing.

**Reply by orangiee** [reply]
hi，lperron
thank you for you reply. but i don't understand, what do u mean 'model is wrong' ? thanks

**Comment by lperron** [comment, marked as answer]
you modified the model. You increased the number of nodes.
Is the demand array of the same size as the distance matrix ?

The current python code of the example expects some dimensions to be equal
in the data part.
Are you sure this is done correctly?

Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by orangiee** [reply]
Yes, you are right, that's the different size which cause the crash! Sorry for the inconvenient and your help in time! 

Thank you so much! I
